### PR TITLE
E2e apps

### PR DESCRIPTION
**Pull request by PauloGoncalvesLima**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Improves the log for failing tests in e2e/apps.

#### Which issue(s) this PR fixes:
Fixes a section of #105678

#### Special notes for your reviewer:
I did the task that was assigned to me at #105678 but it appears #109944 was accepted, so I do not know if this PR is valid.

#### Does this PR introduce a user-facing change?
NONE

### CODE REVIEW

**Comment by aojea** [comment]
/assign @pohly 

as author of the issue https://github.com/kubernetes/kubernetes/issues/105678

**Comment by pohly** [comment]
@PauloGoncalvesLima: sorry for the confusion. I had lost track of https://github.com/kubernetes/kubernetes/pull/109944 when I said in the issue that e2e/apps still needs to be worked on. Your changes now probably conflict with the ones from that other PR, but looking at the latest master there is still at least on (new?) line that needs to be fixed:

```
test/e2e/apps/cronjob.go:               framework.ExpectEqual(apierrors.IsInvalid(err), true, "CronJob creation should fail with invalid time zone error")
```

**Comment by pohly** [comment]
@PauloGoncalvesLima: do you want to cover all remaining issues with your PR?

```
$ git grep 'Expect.*Equal([^,]*, *\(true\|false\)' test/e2e | grep -v test/e2e/framework/log_test.go
test/e2e/apimachinery/crd_conversion_webhook.go:        framework.ExpectEqual(exists, true)
test/e2e/apimachinery/crd_conversion_webhook.go:        framework.ExpectEqual(hostExists, false)
test/e2e/apimachinery/crd_conversion_webhook.go:        framework.ExpectEqual(portExists, false)
test/e2e/apimachinery/crd_conversion_webhook.go:        framework.ExpectEqual(hostPortExists, false)
test/e2e/apimachinery/crd_conversion_webhook.go:        framework.ExpectEqual(hostExists, true)
test/e2e/apimachinery/crd_conversion_webhook.go:        framework.ExpectEqual(portExists, true)
test/e2e/apimachinery/namespace.go:             framework.ExpectEqual(foundFinalizer, true, "Finalizer %q was not found. Namespace %q has %#v", fakeFinalizer, updatedNamespace.Spec.Finalizers)
test/e2e/apimachinery/namespace.go:             framework.ExpectEqual(foundFinalizer, false, "Finalizer %q was found. Namespace %q has %#v", fakeFinalizer, updatedNamespace.Spec.Finalizers)
test/e2e/apps/cronjob.go:               framework.ExpectEqual(apierrors.IsInvalid(err), true, "CronJob creation should fail with invalid time zone error")
test/e2e/scheduling/limit_range.go:             framework.ExpectEqual(checkMinLimitRange, true, "LimitRange does not have the correct min limitRange. Currently is %#v ", patchedLimitRange.Spec.Limits[0].Min)
```

**Comment by PauloGoncalvesLima** [comment]
I can do it @pohly, no problem. Ty

**Comment by PauloGoncalvesLima** [comment]
@pohly, so I did a mistake by doing a merge with the master to my current branch, and now there are 48 file changes, tried to do a revert to my previous commit but with no avail.

**Comment by PauloGoncalvesLima** [comment]
Nice, managed to fix it. Rebase save lives. :smile:

File: test/e2e/apps/rc.go [hunk 1]
```diff
@@ -167,6 +168,9 @@ var _ = SIGDescribe("ReplicationController", func() {
 			if !eventFound {
 				framework.Failf("failed to find RC %v event", watch.Added)
 			}
+			if !eventFound {
+				framework.Failf("failed to find RC %v event", watch.Added)
+			}
```
Comment (EuricoAbreu) [inline test/e2e/apps/rc.go:173]: Repeated code snippet

File: test/e2e/apps/cronjob.go [hunk 2]
```diff
@@ -311,6 +311,9 @@ var _ = SIGDescribe("CronJob", func() {
 		if !apierrors.IsInvalid(err) {
 			framework.Failf("Failed to create CronJob, invalid time zone.")
 		}
+		if !apierrors.IsInvalid(err) {
+			framework.Failf("Failed to create CronJob, invalid time zone.")
+		}
```
Comment (EuricoAbreu) [inline test/e2e/apps/cronjob.go:316]: Repeated code snippet

File: test/e2e/apps/cronjob.go [hunk 3]
```diff
@@ -401,6 +404,9 @@ var _ = SIGDescribe("CronJob", func() {
 				if !isCronJob {
 					framework.Failf("expected CronJob, got %T", evt.Object)
 				}
+				if !isCronJob {
+					framework.Failf("expected CronJob, got %T", evt.Object)
+				}
```
Comment (EuricoAbreu) [inline test/e2e/apps/cronjob.go:409]: Repeated code snippet

**Comment by PauloGoncalvesLima** [comment]
Hello @bart0sh, ok! I'm going to fix it today !!

File: test/e2e/apps/cronjob.go [hunk 4]
```diff
@@ -481,10 +482,7 @@ var _ = SIGDescribe("CronJob", func() {
 		cjs, err = cjClient.List(ctx, metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
 		// Should have <= 2 items since some cronjobs might not have been deleted yet due to finalizers
-		if len(cjs.Items) > 2 {
-			framework.Logf("got unexpected filtered list: %v", cjs.Items)
-			framework.Fail("filtered list should be <= 2")
-		}
+		gomega.Expect(len(cjs.Items)).To(gomega.BeNumerically("<=", 2), "filtered list should be <= 2")
```
Comment (pohly) [inline test/e2e/apps/cronjob.go:485]: The list content is no longer available.

```suggestion
		gomega.Expect(len(cjs.Items)).To(gomega.BeNumerically("<=", 2), "filtered list length should be <= 2, got:\n%s", format.Object(cjs.Items, 1))
```

where "format" is gomega/format.

File: test/e2e/apps/rc.go [hunk 5]
```diff
@@ -269,9 +271,8 @@ var _ = SIGDescribe("ReplicationController", func() {
 			})
 			framework.ExpectNoError(err, "Failed to find updated ready replica count")
 			if !eventFound {
-				framework.Fail("Failed to find updated ready replica count")
+				framework.Failf("Failed to find updated ready replica count")
```
Comment (pohly) [inline test/e2e/apps/rc.go:274]:
```suggestion
				framework.Fail("Failed to find updated ready replica count")
```

File: test/e2e/apps/rc.go [hunk 6]
```diff
@@ -329,7 +330,7 @@ var _ = SIGDescribe("ReplicationController", func() {
 			})
 			framework.ExpectNoError(err, "Wait until condition with watch events should not return an error")
 			if !eventFound {
-				framework.Fail("Failed to find updated ready replica count")
+				framework.Failf("Failed to find updated ready replica count")
```
Comment (pohly) [inline test/e2e/apps/rc.go:333]:
```suggestion
				framework.Fail("Failed to find updated ready replica count")
```

File: test/e2e/apps/rc.go [hunk 7]
```diff
@@ -360,16 +361,15 @@ var _ = SIGDescribe("ReplicationController", func() {
...
 				framework.Failf("failed to find RC %v event", watch.Added)
 			}
 
 			ginkgo.By("listing all ReplicationControllers")
 			rcs, err := f.ClientSet.CoreV1().ReplicationControllers("").List(ctx, metav1.ListOptions{LabelSelector: "test-rc-static=true"})
 			framework.ExpectNoError(err, "failed to list ReplicationController")
-			if len(rcs.Items) == 0 {
-				framework.Fail("Expected to find a ReplicationController but none was found")
-			}
+			gomega.Expect(len(rcs.Items)).To(gomega.BeNumerically(">", 0), "rcs items needs to be > 0.")
```
Comment (pohly) [inline test/e2e/apps/rc.go:372]: I like the original message better.

File: test/e2e/apps/rc.go [hunk 8]
```diff
@@ -382,8 +382,7 @@ var _ = SIGDescribe("ReplicationController", func() {
 				}
 			}
 			if !foundRc {
-				framework.Logf("Got unexpected replication controller list %v", rcs.Items)
-				framework.Failf("could not find ReplicationController %s", testRcName)
+				framework.Failf("ReplicationController doesn't has expected values.")
```
Comment (pohly) [inline test/e2e/apps/rc.go:385]: Also dump the list here.

**Review by pohly — COMMENTED** [review]
Nit: there are also some white space changes that don't seem to be necessary.

File: test/e2e/apps/rc.go (hunk 8 above)
Comment (pohly) [reply test/e2e/apps/rc.go:386]: Dump it inside the failure message... :sweat_smile:

Comment (PauloGoncalvesLima) [reply test/e2e/apps/rc.go:386]: Oh... Ok 😄

File: test/e2e/apps/rc.go [hunk 9]
```diff
@@ -382,8 +382,7 @@ var _ = SIGDescribe("ReplicationController", func() {
 				}
 			}
 			if !foundRc {
-				framework.Logf("Got unexpected replication controller list %v", rcs.Items)
-				framework.Failf("could not find ReplicationController %s", testRcName)
+				framework.Failf("ReplicationController doesn't have expected values. Values that are in the ReplicationController list %v", rcs.Items)
```
Comment (pohly) [inline test/e2e/apps/rc.go:385]: Please format with `format.Object`, as elsewhere.

Comment (PauloGoncalvesLima) [reply test/e2e/apps/rc.go:385]: Ok, Sorry.

File: test/e2e/apps/rc.go [hunk 10]
```diff
@@ -382,8 +383,7 @@ var _ = SIGDescribe("ReplicationController", func() {
 				}
 			}
 			if !foundRc {
-				framework.Logf("Got unexpected replication controller list %v", rcs.Items)
-				framework.Failf("could not find ReplicationController %s", testRcName)
+				framework.Failf("ReplicationController doesn't have expected values.\n Values that are in the ReplicationController list: %v", format.Object(rcs.Items, 1))
```
Comment (pohly) [inline test/e2e/apps/rc.go:386]: Nit:
```suggestion
				framework.Failf("ReplicationController doesn't have expected values.\nValues that are in the ReplicationController list:\n%s", format.Object(rcs.Items, 1))
```

Comment (pohly) [reply test/e2e/apps/rc.go:386]: Please squash all of your commits into one.

**Review by pohly — CHANGES_REQUESTED** [review]
Please squash into a single commit.

**Review by pohly — APPROVED** [review]
/lgtm
/approved

**Comment by pohly** [comment]
Thanks @PauloGoncalvesLima!

**Pull request merged**